docs: setup helpers documentation (#19267)
hashicorp/nomad: website/content/docs/commands/setup/consul.mdx

---
layout: docs
page_title: 'Commands: setup consul'
description: |
  Setup a Consul cluster for Nomad integration.
---

# Command: setup consul

This command sets up Consul for allowing Nomad workloads to authenticate
themselves using Workload Identity.

This command requires `acl:write` permissions for Consul and respects
`CONSUL_HTTP_TOKEN`, `CONSUL_HTTP_ADDR`, and other [Consul-related environment
variables][consulenv].

<Warning>

This command is an experimental feature and may change its behavior in future
versions of Nomad.

</Warning>

## Usage

```plaintext
nomad setup consul [options]
```

## General Options

> @include 'general_options_no_namespace.mdx'

Author: oof, copy-pasta bug.

## Setup Consul Options

- `-jwks-url`: URL of Nomad's JWKS endpoint contacted by Consul to verify JWT
  signatures. Defaults to `http://localhost:4646/.well-known/jwks.json`.

- `-cleanup`: Removes all configuration components this command created from the
  Consul cluster.

- `-y`: Automatically answers "yes" to all the questions, making the setup
  non-interactive. Defaults to "false".

## Examples

> Below is an example of an interactive session with default options, interrupted
> by answering "no" to one of the questions, demonstrating the helper's "cleanup"
> capabilities as well.

Reviewer: ```suggestion
Below is an example of an interactive session with default options, interrupted
by answering `no` to one of the questions, demonstrating the capabilities of the 
`-cleanup` flag.
```
